packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -334,6 +345,13 @@ export class Pipeline extends PipelineBase {
     // @deprecated(v2): switch to default false
     this.crossAccountKeys = props.crossAccountKeys ?? true;
 
+    this.enableKeyRotation = props.enableKeyRotation;
+
+    // Cross account keys must be set for key rotation to be enabled
+    if (props.enableKeyRotation && !props.crossAccountKeys) {
```

Comment:
```suggestion
    if (this.enableKeyRotation && !this.crossAccountKeys) {
```

In your current code, you're not using the defaulting to `true` that happens on line 346 above.